adagrams/game.py

```diff
@@ -1,11 +1,159 @@
+import random
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+
 def draw_letters():
-    pass
+
+    my_list = [] #creating an empty list to collect randomly picked letter
+
+    #converting dictionary keys into list representing the 'count' times of letter and storing in pool_list
+    pool_list = []
+    for letter, count in LETTER_POOL.items():
+        pool_list += [letter] * count # add letter 'count' times to pool_list
+        
+    while len(my_list) < 10:
+        random_key = random.randint(0, len(pool_list) - 1) #randomly select index from pool_list
+        
+        random_letter = pool_list[random_key] #getting letter associated with index
+        pool_list.remove(random_letter) #removing letter from pool_list
+        my_list.append(random_letter) #adding this letter to my_list
+
+    return my_list
+
+
+letter_bank = draw_letters() # retreiving my_list of 10 listter from draw_letter function
+
+word = "SANTAAA"
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+
+    # getting user input as string 
+    upper_word = word.upper()
+
+    letter_bank_dict = {} #empty dictionary to store 
+    inputed_word_dict = {} #empty dictionary to store letter and frequency from user
+
+
+    # adding upper_word into inputed_word_dict with letter: count format
+    for letter in upper_word:
+        if letter not in inputed_word_dict:
+            inputed_word_dict[letter] = 1 #Initialize with one if letter added first time
+        else:
+            inputed_word_dict[letter] += 1 #Increment by 1 if letter added additionally
+    
+
+    # converting answer_list to a letter_bank_dict with a format of letter:count
+    for letter in letter_bank:
+        if letter not in letter_bank_dict:
+            letter_bank_dict[letter] = 1
+        else:
+            letter_bank_dict[letter] += 1
+
+
+    # comparing letter_bank_dict and inputed_word_dict 
+    for key, value in inputed_word_dict.items():
+        if key not in letter_bank_dict or value > letter_bank_dict[key]:  # checking if key from inputed_word_dict is in letter_bank_dict and checking if value of inputed_word_dict is greater than value in letter_bank_dict
+            return False
+        
+    return True
```

Comment:
You create two dictionaries and then use a loop to iterate over them to see if your word is valid for the random letters in a hand. 

Another approach to implementing `uses_available_letters` could be to make a copy of the list that represents the randomly drawn words. If a letter is in the word and in the hand, then remove the letter from the hand. If the letter is not in the hand then you can immediately return False. If the loop never returns False, then you know that you had a valid word for the hand you have. This would reduce the complexity of your solution by only using one additional data structure and is a little more readable.

```python
# use list comprehension to make a copy
copied_letter_bank = [letter for letter in letter_bank]

# or use list slicing to make a copy (either way works)
# copied_letter_bank = letter_bank[:]

for letter in word.upper():
        if letter in copied_letter_bank:
            copied_letter_bank.remove(letter)
        else:
            return False

    return True
```